src/main/java/ca/on/oicr/gsi/dimsum/service/CaseService.java

```diff
@@ -550,6 +497,22 @@ private Stream<TestTableView> filterTestTableViews(List<Case> cases,
     return stream;
   }
 
+  private Map<Case, List<Test>> getFilteredCaseAndTest(List<Case> cases,
+      Collection<CaseFilter> filters) {
+    Map<Case, List<Test>> testsByCase = filterCases(cases, filters)
+        .collect(Collectors.toMap(Function.identity(), x -> x.getTests()));
```

Comment:
No need for this to be a `Map` here. You can iterate over the `List` and use `Case::getTests` instead of `entry.getValue()` below

Reply:
You do need to create that map later. Currently, you are creating this map that contains all of the tests, and then you're replacing all of the values later. I'm suggesting you wait until later and create the map with the correct values initially

Reply:
Ah I see what you mean. Thank you